lib/common/strings.c

```diff
@@ -412,6 +412,57 @@ crm_get_msec(const char *input)
     return msec;
 }
 
+/*!
+ * \brief Parse milliseconds from a Pacemaker interval specification
+ *
+ * \param[in]  input      Pacemaker time interval specification (a bare number
+ *                        of seconds; a number with a unit, optionally with
+ *                        whitespace before and/or after the number; or an ISO
+ *                        8601 duration)
+ * \param[out] result_ms  Where to store milliseconds equivalent of \p input on
+ *                        success (limited to the range of an unsigned integer),
+ *                        or 0 if \p input is \c NULL or invalid
+ *
+ * \return Standard Pacemaker return code (specifically, \c pcmk_rc_ok if
+ *         \p input is valid or \c NULL, and \c EINVAL otherwise)
+ */
+int
+pcmk__parse_interval_spec(const char *input, guint *result_ms)
```

Comment:
I can do that. I didn't see much need since for anything an external program is likely to need to parse, Pacemaker has already converted it to milliseconds or seconds (depending on the context)... unless `pcmk__parse_interval_spec()` can handle the output of `pcmk__readable_interval()`. That just occurred to me.

I felt like exposing more of Pacemaker's generic time representation stuff (much like its generic XML or string stuff) was a bad idea. I don't feel that strongly about it though.

Reply:
I agree we shouldn't expose generic time handling, but interval specs are Pacemaker-specific CIB syntax, so external code might reasonably want to parse it. The readable interval format isn't the same, but that might have been a good idea.